pylint/checkers/utils.py

```diff
@@ -1362,11 +1368,21 @@ def safe_infer(
     if value is not astroid.Uninferable:
         inferred_types.add(_get_python_type_of_node(value))
 
+    first_constant_value = None
+    if compare_constants and isinstance(value, nodes.Const):
+        first_constant_value = value.value
+
     try:
         for inferred in infer_gen:
             inferred_type = _get_python_type_of_node(inferred)
             if inferred_type not in inferred_types:
                 return None  # If there is ambiguity on the inferred node.
+            if (
+                compare_constants
+                and isinstance(inferred, nodes.Const)
+                and inferred.value != first_constant_value
+            ):
+                return None
```

Comment:
```suggestion
            if (
                compare_constants
                and isinstance(inferred, nodes.Const)
                and isinstance(value, nodes.Const)
                and inferred.value != value.value
            ):
                return None
```

I think you could write it like this and remove the logic above since `value` isn't mutated.